src/Statement/StatementList.php

```diff
@@ -87,7 +87,13 @@ public function getPropertyIds() {
 	}
 
 	public function addStatement( Statement $statement ) {
-		$this->statements[] = $statement;
+		if ( $statement->getGuid() !== null
+			&& !array_key_exists( $statement->getGuid(), $this->statements )
+		) {
+			$this->statements[$statement->getGuid()] = $statement;
+		} else {
+			$this->statements[] = $statement;
```

Comment:
You can't say that a class which does indexing by GUID does not care about GUIDs, or uses statements "o matter what their GUID is".